MonoGame.Framework/Platform/SDL/SDLGameWindow.cs

```diff
@@ -302,7 +302,20 @@ public void ClientResize(int width, int height)
 
         public void CallTextInput(char c, Keys key = Keys.None)
         {
-            OnTextInput(this, new TextInputEventArgs(c, key));
+            if(IsTextInputHandled)
```

Comment:
You don't need these checks. `EventHelpers.Raise` does the null check.
I suggest we make the added EventArgs structs so there's no allocation here.

Reply:
Its not a null check, its there to avoid garbage.

Reply:
InputKeyEventArgs inherits from EventArgs, which is a requirement for the event machinery. So, I can't convert this to a structure.

Reply:
It's not a requirement. Usually event arguments inherit from it in case consumers don't care about the argument.

Reply:
Here's what happens when I remove that inheritance

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0311	The type 'Microsoft.Xna.Framework.InputKeyEventArgs' cannot be used as type parameter 'TEventArgs' in the generic type or method 'EventHelpers.Raise<TEventArgs>(object, EventHandler<TEventArgs>, TEventArgs)'. There is no implicit reference conversion from 'Microsoft.Xna.Framework.InputKeyEventArgs' to 'System.EventArgs'.	MonoGame.Framework.Windows	C:\Users\JKiesel.CISDEV\MonoGame\MonoGame.Framework\GameWindow.cs	172	Active
```

Reply:
Ok, I think I found a way to get the best of both worlds. I'm bypassing the EventHelper and invoking the event directly with my own null check.

Reply:
Ah, right. The EventHelper that's used has a generic type constraint so type arguments must inherit from EventArgs. We should get rid of that so we can use structs IMO.

Reply:
@Jjagg just so we have clear expectations, is that a change you want me to make here?

Reply:
Yes, please do :)